dispatch/apps/events/views.py

```diff
@@ -13,16 +13,39 @@ def submit_success(request):
     return render(request, 'events/submit/success.html')
 
 def submit_form(request):
-    facebook_url = request.POST.get('facebook_url')
-    facebook_error = False
+    event_url = request.POST.get('event_url')
+    url_error = False
+
+    if request.POST.get('url_import') and event_url is not None:
+
+        handlers = {
+            'calendar.events.ubc.ca': {
+                'url_handler': UBCEvent,
+                'url_handler_error': UBCEventError
+            },
+            'facebook.com/events/': {
```

Comment:
I like this! I am going to keep the actual data collection from the handler wrapped in a try-except to catch any errors that occur in that process